ci/include-grouping.py

```diff
@@ -0,0 +1,159 @@
+                    includes.append(line)
+                    skipped_lines = []
+                else:
+                    if len(includes) > 0:
+                        # regroup all includes by skipping all empty lines while scanning
+                        if len(line.strip()) == 0:
+                            skipped_lines.append(line)
+                            continue
+                        print_sorted_includes(includes, file=out_file)
+                        includes = []
+
+                    # print skipped lines
+                    for skipped_line in skipped_lines:
+                        print(skipped_line, end="", file=out_file)
+                    skipped_lines = []
+
+                    print(line, end="", file=out_file)
+
+    # move out_file to file
+    os.rename(file + ".tmp", file)
+
+    pass
+
+
+def print_sorted_includes(includes, file=sys.stdout):
+    default_group_key = 1
+    groups = {}
+
+    matches = [
+        {
+            "name": "system includes",
+            "matcher": lambda x: x.startswith("<"),
+            "group_key": -2,
+            "priority": 0,
+        },
+        {
+            "name": "naked postgres includes",
+            "matcher": lambda x: "/" not in x,
+            "group_key": 0,
+            "priority": 9,
+        },
+        {
+            "name": "postgres.h",
+            "list": ['"postgres.h"'],
+            "group_key": -1,
+            "priority": -1,
+        },
+        {
+            "name": "naked citus inlcudes",
+            "list": ['"citus_version.h"', '"pg_version_compat.h"'],
+            "group_key": 3,
+            "priority": 0,
+        },
+        {
+            "name": "positional citus includes",
+            "list": ['"distributed/pg_version_constants.h"'],
+            "group_key": 4,
+            "priority": 0,
+        },
+        {
+            "name": "columnar includes",
+            "matcher": lambda x: x.startswith('"columnar/'),
+            "group_key": 5,
+            "priority": 1,
+        },
+        {
+            "name": "distributed includes",
+            "matcher": lambda x: x.startswith('"distributed/'),
+            "group_key": 6,
+            "priority": 1,
+        },
+    ]
+
+    matches.sort(key=lambda x: x["priority"])
+
+    common_system_include_error_prefixes = ["<nodes/", "<distributed/"]
+
+    for include in includes:
+        # extract the group key from the include
+        include_content = include.split(" ")[1]
+
+        # fix common system includes which are secretly postgres or citus includes
+        for common_prefix in common_system_include_error_prefixes:
+            if include_content.startswith(common_prefix):
+                include_content = '"' + include_content.strip()[1:-1] + '"'
+                include = include.split(" ")[0] + " " + include_content + "\n"
+                break
+
+        group_key = default_group_key
+        for matcher in matches:
+            if "list" in matcher:
```

Comment:
Do we need to make the difference between "matcher" and "list"?
The last one could also be written as "matcher":
```python
"matcher": lambda x: x in ['"citus_version.h"', '"pg_version_compat.h"'],
```

It will allow to get rid of unnecessary branching.